gem/tests/test_auth.py

```diff
@@ -441,3 +447,291 @@ def test_auth_middleware(self):
+            password='pass'
+        )
+        request = RequestFactory().post(
+            data={
+                'username': user.username,
+                'password': user.password
+            }, path=reverse('wagtailadmin_login'))
+        request.site = self.site
+        self.assertFalse(adaptor.is_open_for_signup(request, None))
+
+    def test_staff_user_adapter_staff_user_perms(self):
+        adaptor = StaffUserAdapter()
+        group = Group.objects.filter().first()
+        perm = Permission.objects.filter().first()
+        user = get_user_model().objects.create(
+            username='testuser',
+            email='testuser@email.com',
+            is_staff=True,
+            password='pass'
+        )
+        user.groups.add(group)
+        user.user_permissions.add(perm)
+        request = RequestFactory().post(
+            data={
+                'username': user.username,
+                'password': user.password
+            }, path=reverse('wagtailadmin_login'))
+        request.site = self.site
+        self.assertFalse(adaptor.is_open_for_signup(request, None))
+
+    def test_user_delete(self):
+        user = get_user_model().objects.create(
+            username='testuser',
+            email='testuser@email.com',
+            is_staff=True,
+            password='pass'
+        )
+        SocialAccount.objects.create(
+            user=user, provider='google', uid='1')
+        user.delete()
+        self.assertFalse(
+            SocialAccount.objects.filter(user=user)
+        )
+
+    @override_settings(ENABLE_ALL_AUTH=True)
+    def test_invite_create_view(self):
+        req = RequestFactory()
+        req.site = self.site
+        req.user = self.user
+
+        self.client.force_login(self.user)
+        url = '/admin/gem/invite/create/'
+        data = {
+            'email': 'testinvite@test.com'
+        }
+        res = self.client.post(url, data=data, request=req)
+
+        subject = '{}: Admin site invitation'.format('localhost [default]')
+        self.assertEqual(res.status_code, 302)
+
+        self.assertEqual(len(mail.outbox), 1)
+        self.assertEqual(mail.outbox[0].subject, subject)
+        self.assertEqual(mail.outbox[0].to, [data['email']])
+        self.assertEqual(mail.outbox[0].from_email, 'no-reply@gehosting.org')
+
+        self.assertTrue(
+            Invite.objects.filter(user=self.user).exists())
+
+    @override_settings(ENABLE_ALL_AUTH=True)
+    def test_invite_edit_view(self):
+        data = {
+            'email': 'testinvite@test.com'
+        }
+        req = RequestFactory()
+        req.site = self.site
+        req.user = self.user
+
+        invite = Invite.objects.create(
+            email=data['email'], user=self.user, site=self.site)
+        self.client.force_login(self.user)
+
+        url = '/admin/gem/invite/edit/{}/'.format(invite.pk)
+        res = self.client.post(url, request=req)
+        self.assertEqual(res.status_code, 200)
```

Comment:
Please test that the existing invite is displayed or that the correct template is being used here.